autoload/ale/engine.vim

```diff
@@ -108,6 +108,9 @@ function! s:HandleExit(job) abort
         call ale#sign#SetSigns(l:buffer, g:ale_buffer_loclist_map[l:buffer])
     endif
 
+    " Call user autocommands. This allows users to hook into ALE's lint cycle.
+    doautocmd User ALELint
```

Comment:
I totally approve of this.

Could you document this behaviour in the documentation? I'll make a note to write a test which checks that the autocmd event is fired.